rdr_service/dao/hpro_consent_dao.py

```diff
@@ -0,0 +1,44 @@
+
+from rdr_service.dao.base_dao import UpdatableDao
+from rdr_service.model.consent_file import ConsentFile
+from rdr_service.model.hpro_consent_files import HealthProConsentFile
+
+
+class HealthProConsentDao(UpdatableDao):
+    validate_version_match = False
+
+    def __init__(self):
+        super(HealthProConsentDao, self).__init__(
+            HealthProConsentFile, order_by_ending=['id'])
+
+    def get_needed_consents_for_transfer(self, limit=None):
+        with self.session() as session:
+            needed_consents = session.query(
+                ConsentFile
+            ).outerjoin(
+                HealthProConsentFile,
+                HealthProConsentFile.consent_file_id == ConsentFile.id
+            ).filter(
+                ConsentFile.file_exists == 1,
+                ConsentFile.file_path.isnot(None),
+                HealthProConsentFile.id.is_(None)
+            )
```

Comment:
Was part of the use case for this effort that the end users could visually inspect/see issues with the consent files for themselves? But, unsure if they would also want to flag the file in the front end as having an issue and if we need a way to communicate that status to HPRO

Reply:
I would look at the `sync_status` value. If it's `READY_FOR_SYNC` or `SYNC_COMPLETE` then it's valid and should be sent. Otherwise it should be left alone until it changes (if it does).